rasa_core/policies/sklearn_policy.py

```diff
@@ -139,44 +154,57 @@ def train(self, training_data, domain, **kwargs):
         if score is not None:
             logger.info("Cross validation score: {:.5f}".format(score))
 
-    def continue_training(self, training_data, domain, **kwargs):
-        X, y = self._extract_training_data(training_data)
-        Xt, yt = self._preprocess_data(X, y)
-        if not hasattr(self.model, 'partial_fit'):
-            raise TypeError("Continuing training is only possible with "
-                            "sklearn models that support 'partial_fit'.")
-        self.model.partial_fit(Xt, yt)
+    def _postprocess_prediction(self, y_proba, domain):
+        yp = y_proba[0].tolist()
+
+        # Some classes might not be part of the training labels. Since
+        # sklearn does not predict labels it has never encountered
+        # during training, it is necessary to insert missing classes.
+        indices = self.label_encoder.inverse_transform(np.arange(len(yp)))
+        y_filled = [0.0 for _ in range(domain.num_actions)]
+        for i, pred in zip(indices, yp):
+            y_filled[i] = pred
+
+        return y_filled
 
     def predict_action_probabilities(self, tracker, domain):
-        X_feat = self.featurize(tracker, domain)
-        Xt = self._preprocess_data(X_feat[np.newaxis])
+        # type: (DialogueStateTracker, Domain) -> List[float]
+        X = self.featurizer.create_X([tracker], domain)
+        Xt = self._preprocess_data(X)
         y_proba = self.model.predict_proba(Xt)
-        return self._postprocess_prediction(y_proba)
+        return self._postprocess_prediction(y_proba, domain)
 
     def persist(self, path):
-        if not self.model:
+        # type: (Text) -> None
+
+        if self.model:
+            self.featurizer.persist(path)
+
+            filename = os.path.join(path, 'sklearn_model.pkl')
+            with open(filename, 'wb') as f:
+                pickle.dump(self._state, f)
+        else:
             warnings.warn("Persist called without a trained model present. "
                           "Nothing to persist then!")
-            return
-
-        filename = os.path.join(path, 'sklearn_model.pkl')
-        with open(filename, 'wb') as f:
-            pickle.dump(self._state, f)
 
     @classmethod
-    def load(cls, path, featurizer, max_history):
+    def load(cls, path):
+        # type: (Text) -> Policy
         filename = os.path.join(path, 'sklearn_model.pkl')
         if not os.path.exists(path):
             raise OSError("Failed to load dialogue model. Path {} "
                           "doesn't exist".format(os.path.abspath(filename)))
 
+        featurizer = TrackerFeaturizer.load(path)
+        assert isinstance(featurizer, MaxHistoryTrackerFeaturizer), \
+            ("Loaded featurizer of type {}, should be "
+             "MaxHistoryTrackerFeaturizer.".format(type(featurizer).__name__))
+
+        policy = cls(featurizer=featurizer)
+
         with open(filename, 'rb') as f:
             state = pickle.load(f)
+        vars(policy).update(state)
```

Comment:
does this work? I think that dict is write protected for new style classes? but yeah you didn't write that - just noticed

Reply:
that's i do not know, we need to check

Reply:
don't worry about it now, it's nothing this PR changed.